src/engines/caching.cc

```diff
@@ -0,0 +1,315 @@
+/*
+ * Copyright 2017-2018, Intel Corporation
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in
+ *       the documentation and/or other materials provided with the
+ *       distribution.
+ *
+ *     * Neither the name of the copyright holder nor the names of its
+ *       contributors may be used to endorse or promote products derived
+ *       from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <iostream>
+#include "caching.h"
+#include "kvtree3.h"
+#include "btree.h"
+#include "vmap.h"
+#include "vcmap.h"
+#include <lib_acl.hpp>
+#include <lib_acl.h>
+#include <stream/socket_stream.hpp>
+#include <redis/redis_stream.hpp>
+#include <redis/redis_client.hpp>
+#include <redis/redis_result.hpp>
+#include <list>
+#include <libmemcached/memcached.h>
+#include <string>
+
+#define DO_LOG 0 
+#define LOG(msg) if (DO_LOG) std::cout << "[caching] " << msg << "\n"
+#define ZERO 0
+
+namespace pmemkv {
+namespace caching {
+
+CachingEngine::CachingEngine(const string& config) {
+    if (!readConfig(config) || !(basePtr = KVEngine::Start(subEngine, subEngineConfig)))
+       throw "Caching Engine Exception"; 
```

Comment:
Remove the space between Caching and Engine so it's:
throw "CachingEngine Exception";